verification/epic.go

```diff
@@ -0,0 +1,163 @@
+package verification
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/golang/glog"
+
+	"github.com/livepeer/go-livepeer/common"
+
+	"github.com/livepeer/lpms/ffmpeg"
+)
+
+var ErrVideoUnavailable = errors.New("VideoUnavailable")
+var ErrAudioMismatch = Retryable{errors.New("AudioMismatch")}
+var ErrTampered = Retryable{errors.New("Tampered")}
+
+type epicResolution struct {
+	Width  int `json:"width"`
+	Height int `json:"height"`
+}
+type epicRendition struct {
+	URI        string         `json:"uri"`
+	Resolution epicResolution `json:"resolution"`
+	Framerate  uint           `json:"frame_rate"`
+	Pixels     int64          `json:"pixels"`
+}
+type epicRequest struct {
+	Source         string          `json:"source"`
+	Renditions     []epicRendition `json:"renditions"`
+	OrchestratorID string          `json:"orchestratorID"`
+	Model          string          `json:"model"`
+}
+
+type epicResults struct {
+	Source  string `json:"source"`
+	Results []struct {
+		VideoAvailable bool    `json:"video_available"`
+		AudioAvailable bool    `json:"audio_available"`
+		AudioDistance  float64 `json:"audio_dist"`
+		Pixels         int64   `json:"pixels"`
+		Tamper         float64 `json:"tamper"`
+	} `json:"results"`
+}
+
+type verificationResult struct {
+	score  float64
+	pixels []int64
+}
+
+type EpicClassifier struct {
+	Addr string
+}
+
+func (vr *verificationResult) Pixels() []int64 {
+	return vr.pixels
+}
+func (vr *verificationResult) Score() float64 {
+	return vr.score
+}
+
+func epicResultsToVerificationResults(er *epicResults) (*verificationResult, error) {
+	// find average of scores and build list of pixels
```

Comment:
Ah so an implication of our current transcoding workflow where we either receive all renditions requested or we consider the segment transcoding as failed is that if 1 out of N renditions is classified as tampered we retry for all N renditions. I feel like ideally in that case we only retry for the single rendition, but I get why retrying for all N renditions fits better with our current workflow.